solver.rb

```diff
@@ -0,0 +1,31 @@
+class Solver
+  def factorial(number)
+    if number.zero?
+      1
+    elsif number.negative?
+      raise ArgumentError, 'You entered a Negative number Please use only Positive.'
+    else
+      number * factorial(number - 1)
+    end
+  end
+
+  def reverse(string)
+    if string == ''
+      ''
+    else
```

Comment:
* Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.  currently your commit history show otherwise, the idea is to write tests first and then the code. This will be verified in your commit history, so it is important to use the proper sequence. Kindly fix this [Check project requirement here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=Remember%20to%20start%20with%20tests%20for%20all%20of%20the%20methods%20described%20above.%20Your%20commit%20history%20should%20prove%20that%20you%20have%20used%20TDD.)